app/models/maintenance_tasks/csv_collection.rb

```diff
@@ -13,6 +13,18 @@ module CsvCollection
     # @return [String] the content of the CSV file to process.
     attr_accessor :csv_content
 
+    # @api private
+    CsvEnumeratorBuilder = Struct.new(:csv) do
+      def enumerator(context:)
+        JobIteration::CsvEnumerator.new(csv).rows(cursor: context.cursor)
+      end
+    end
+    private_constant :CsvEnumeratorBuilder
+
+    def enumerator_builder
+      CsvEnumeratorBuilder.new(collection)
+    end
```

Comment:
Since this module exists, it felt like the natural place to put the CSV enumerator code, instead of directly in `Task#enumerator_builder`.